File: RailgunVB/Commands/Cat.vb
Imports Discord.Commands
Imports RailgunVB.Core.Api

Namespace Commands
    
    <Group("cat")>
    Public Class Cat
        Inherits ModuleBase
    
        Private ReadOnly _randomCat As RandomCat

        Public Sub New(randomCat As RandomCat)
            _randomCat = randomCat
        End Sub
        
        <Command>
        Public Async Function CatAsync() As Task
            await Context.Channel.SendFileAsync(await _randomCat.GetRandomCatAsync(), "CatImg.png")
        End Function
        
    End Class
    
End NameSpace

File: Railgun/Commands/Cat.cs
using System.Threading.Tasks;
using Discord;
using Finite.Commands;
using Railgun.Apis.RandomCat;
using Railgun.Core;
using Railgun.Core.Attributes;

namespace Railgun.Commands
{
    [Alias("cat"), BotPerms(ChannelPermission.AttachFiles)]
    public class Cat : SystemBase
    {
        private readonly RandomCat _randomCat;

        public Cat(RandomCat randomCat)
            => _randomCat = randomCat;
        
        [Command]
        public async Task ExecuteAsync() 
            => await Context.Channel.SendFileAsync(await _randomCat.GetRandomCatAsync(), "CatImg.png");
    }
}